src/president/game.js

```diff
@@ -0,0 +1,80 @@
+const Hand = require ('./hand.js');
+const Deck = require ('./deck.js');
+
+/*
+ * Create a game object.
+ */
+exports.initGame = function (id, players) {
+    var deck = Deck.initDeck(Deck.deck_json);
+    
+    var game = {
+        id: id, //unique identifier
+        players: players, //array of players names
+        hands: [], //array of players hands
+        // TODO: Current player depends on the rule
+        current_pl: 0, //index of the current player
+        deck: deck //deck of the game
+    }
+    Deck.mixDeck(game.deck);
+    
+    for(var i=0; i<game.players.length; i++) {
+        game.hands[i] = Hand.emptyHand();
+    }
+
+    return game;
+}
+
+/*
+ * Distribute the cards to each player.
+ */
+exports.distributeCards = function (game) {
+    var ind = 0;
+    while(!Deck.isEmptyDeck(game.deck)){
+        var card = Deck.popCard(game.deck);
+        Hand.addCardToHand(game.hands[ind], card);
+        ind = (++ind)%game.players.length;
+    }
+}
+
+/*
+ * Return true if the game is finished
+ */
+exports.finishGame = function (game) {
+    var players_left = game.players.length;
+    
+    game.hands.forEach(hand => {
+        if(Hand.lengthHand(hand) == 0) {
+            players_left--;
+        }
+    });
+
+    return players_left <= 1;
+}
+
+/*
+ * Allows the current playr to play his turn.
+ */
+exports.playOneTurn = function (game) {
+    throw "TODO";
+}
+
+/*
+ * Player plays cards, remove them from his hands.
+ */
+exports.playCards = function (game, player, cards) {
+    var index = game.players.indexOf(player);
+    if(index < 0) {
+        return;
+    }
+    Hand.removeCardsFromHand(game.hands[index], cards);
+}
+
+/*
+ * Play the game until the game has finished.
+ */
+exports.playGame = function (game) {
```

Comment:
As we will use asynchronous functions with *socket.io*, I'm not sure we need this function. (It's a question).

Reply:
This function is definitely temporary, it's just a matter of testing locally the game with a single user doing all the players.

Reply:
Ok :D